cmd/access/cluster.go

```diff
@@ -0,0 +1,446 @@
+		return c.createLocalKubeconfigAccess(cluster, kubeconfigSecret)
+	}
+}
+
+// createJumpPodAccess grants access to a cluster by creating a pod for users to exec into
+func (c *clusterAccessOptions) createJumpPodAccess(cluster *clustersmgmtv1.Cluster, kubeconfigSecret corev1.Secret) error {
+	c.Println("Attempting to spin up a pod to use for access")
+
+	pod, err := c.createJumpPod(kubeconfigSecret, cluster.ID())
+	if err != nil {
+		c.Errorln("Failed to create pod")
+		return err
+	}
+
+	c.Println(fmt.Sprintf("Jump pod created. Waiting for it to start"))
+	c.Println("")
+
+	err = c.waitForJumpPod(pod, jumpPodPollInterval, jumpPodPollTimeout)
+	if err != nil {
+		c.Errorln("Timed out waiting for pod to start.")
+		c.Println(fmt.Sprintf("You can check the status of the pod using\n\n    oc describe pods %s -n %s\n", pod.Name, pod.Namespace))
+		c.Println("Once the pod is running:")
+	} else {
+		c.Println("Pod detected as running")
+	}
+	c.Println(fmt.Sprintf("Use \n\n    oc exec -it --as %s -n %s %s -- /bin/bash\n\nto run commands in the pod. All 'oc' commands run within the pod will be executed against the cluster '%s' (this can be verified by running `oc cluster-info` in the pod)", impersonateUser, pod.Namespace, pod.Name, cluster.Name()))
+	return err
+}
+
+// createLocalKubeconfigAccess grants access to a cluster by writing the cluster's kubeconfig file to the local filesystem and (optionally) updating the user's cli environment
+func (c *clusterAccessOptions) createLocalKubeconfigAccess(cluster *clustersmgmtv1.Cluster, kubeconfigSecret corev1.Secret) error {
+	c.Println("Retrieving kubeconfig secret from Hive")
+
+	kubeconfigFilePath := fpath.Join(os.TempDir(), kubeconfigSecret.Name)
+	rawKubeconfig, found := kubeconfigSecret.Data[kubeconfigSecretKey]
+	if !found {
+		// Kubeconfig secret doesn't contain the expected key - write the obtained secret to a temp location so the user can troubleshoot or manually parse
+		c.Errorln(fmt.Sprintf("\nExpected key '%s' not found in Secret", kubeconfigSecretKey))
+		c.Println("Attempting to save Secret locally")
+
+		rawData, err := json.Marshal(kubeconfigSecret)
+		if err != nil {
+			c.Errorln("Failed to marshal secret to raw data")
+			return err
+		}
+
+		err = saveAsLocalFile(rawData, kubeconfigFilePath)
+		if err != nil {
+			c.Errorln("Failed to write Secret to file")
+			return err
+		}
+
+		c.Println(fmt.Sprintf("File has been written to '%s' for manual use", kubeconfigFilePath))
+		return fmt.Errorf("Could not parse cluster's kubeconfig Secret")
+	}
+
+	// Determine if cluster utilizes a Private API
+	listening, listeningOK := cluster.API().GetListening()
+	if !listeningOK {
+		c.Errorln("\nFailed to determine if the cluster is private.\nIf you're not able to access the cluster, try modifying the resulting kubeconfig according to the SOP: https://github.com/openshift/ops-sop/blob/master/v4/howto/break-glass-kubeadmin.md#for-clusters-with-private-api")
```

Comment:
is return needed?

Reply:
It could be, but I've attempted to make this code as reasonably resistant to failure as possible. In general, this command is only run when things are already going poorly in a cluster or our ops environment, and usually as a last resort. 

With that in mind, I've take the stance in several places that if an error isn't completely fatal, we should just print the error, while still giving a best effort to complete the task at hand. Here, for instance, even if we don't know whether the cluster is private or not, we might still be able to pull it's kubeconfig, which is a requirement for both private & non-private clusters. If the kubeconfig pulled doesn't work, the error message indicates they may have to modify the file, per the SOP, but it at least saves them the effort of having to manually pull the file when we could still do so easily here :). 

(This is also the case when we fail to retrieve the kubeconfig file from within it's secret - we just save the secret locally, that way the SRE can try and manually pull it).

That being said, all of this is very non-obvious, and I definitely should have at least commented something indicating the intentional omission of a return statement here. I've added one now that hopefully clears up confusion in the future.